--- keyboard.js.orig	2010-12-13 21:42:45.000000000 -0800
+++ keyboard.js	2010-12-13 21:27:56.000000000 -0800
@@ -72,7 +72,7 @@
   this.VKI_size = 2;  // Default keyboard size (1-5)
   this.VKI_sizeAdj = true;  // Allow user to adjust keyboard size
   this.VKI_clearPasswords = false;  // Clear password fields on focus
-  this.VKI_imageURI = "keyboard.png";  // If empty string, use imageless mode
+  this.VKI_imageURI = owm_vki_imageuri;  // If empty string, use imageless mode
   this.VKI_clickless = 0;  // 0 = disabled, > 0 = delay in ms
   this.VKI_keyCenter = 3;
 
@@ -84,21 +84,7 @@
   this.VKI_isMoz = (!this.VKI_isWebKit && navigator.product == "Gecko");
 
   /* ***** i18n text strings ************************************* */
-  this.VKI_i18n = {
-    '00': "Display Number Pad",
-    '01': "Display virtual keyboard interface",
-    '02': "Select keyboard layout",
-    '03': "Dead keys",
-    '04': "On",
-    '05': "Off",
-    '06': "Close the keyboard",
-    '07': "Clear",
-    '08': "Clear this input",
-    '09': "Version",
-    '10': "Decrease keyboard size",
-    '11': "Increase keyboard size"
-  };
-
+  this.VKI_i18n = owm_vki_i18n;
 
   /* ***** Create keyboards ************************************** */
   this.VKI_layout = {};
