diff -ruN CGI-SpeedyCGI-2.22.orig/README CGI-SpeedyCGI-2.22/README
--- CGI-SpeedyCGI-2.22.orig/README	Sun Oct 12 12:25:40 2003
+++ CGI-SpeedyCGI-2.22/README	Mon Jun  7 23:30:10 2004
@@ -218,7 +218,7 @@

     TmpBase
             Command Line    : -T<string>
-            Default Value   : "/tmp/speedy"
+            Default Value   : "/var/run/speedy"
             Context         : mod_speedycgi, speedy

             Description:
diff -ruN CGI-SpeedyCGI-2.22.orig/README.html CGI-SpeedyCGI-2.22/README.html
--- CGI-SpeedyCGI-2.22.orig/README.html	Sun Oct 12 12:25:41 2003
+++ CGI-SpeedyCGI-2.22/README.html	Mon Jun  7 23:32:27 2004
@@ -361,7 +361,7 @@
 <DT><STRONG><A NAME="item_TmpBase">TmpBase</A></STRONG><DD>
 <P>
 <PRE>    Command Line    : -T&lt;string&gt;
-    Default Value   : &quot;/tmp/speedy&quot;
+    Default Value   : &quot;/var/run/speedy&quot;
     Context         : mod_speedycgi, speedy
 </PRE>
 <P>
diff -ruN CGI-SpeedyCGI-2.22.orig/lib/CGI/SpeedyCGI.pm CGI-SpeedyCGI-2.22/lib/CGI/SpeedyCGI.pm
--- CGI-SpeedyCGI-2.22.orig/lib/CGI/SpeedyCGI.pm	Sun Oct 12 12:25:41 2003
+++ CGI-SpeedyCGI-2.22/lib/CGI/SpeedyCGI.pm	Mon Jun  7 23:31:43 2004
@@ -388,7 +388,7 @@
 =item TmpBase

     Command Line    : -T<string>
-    Default Value   : "/tmp/speedy"
+    Default Value   : "/var/run/speedy"
     Context         : mod_speedycgi, speedy

     Description:
diff -ruN CGI-SpeedyCGI-2.22.orig/speedy_dump/SpeedyFile.pl CGI-SpeedyCGI-2.22/speedy_dump/SpeedyFile.pl
--- CGI-SpeedyCGI-2.22.orig/speedy_dump/SpeedyFile.pl	Fri Sep 27 02:43:01 2002
+++ CGI-SpeedyCGI-2.22/speedy_dump/SpeedyFile.pl	Mon Jun  7 23:30:00 2004
@@ -12,7 +12,7 @@

 sub fname { my $self = shift;
     $self->{fname} ||=
-	sprintf("%s.${FILEREV}.%x.F", ($ENV{SPEEDY_TMPBASE} || '/tmp/speedy'), $>);
+	sprintf("%s.${FILEREV}.%x.F", ($ENV{SPEEDY_TMPBASE} || '/var/run/speedy'), $>);
 }

 sub data { my $self = shift;
diff -ruN CGI-SpeedyCGI-2.22.orig/src/optdefs CGI-SpeedyCGI-2.22/src/optdefs
--- CGI-SpeedyCGI-2.22.orig/src/optdefs	Tue Oct  7 12:03:48 2003
+++ CGI-SpeedyCGI-2.22/src/optdefs	Mon Jun  7 23:32:11 2004
@@ -69,7 +69,7 @@
 option	TmpBase
 type	str
 letter	T
-defval	/tmp/speedy
+defval	/var/run/speedy
 context	frontend
 desc	Use the given prefix for creating temporary files.  This must be
 desc	a filename prefix, not a directory name.
