diff -ruN suidperl/openwebmail-abook.pl speedycgi/openwebmail-abook.pl
--- suidperl/openwebmail-abook.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-abook.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_abook
 #
 # openwebmail-abook.pl - address book program
 #
diff -ruN suidperl/openwebmail-advsearch.pl speedycgi/openwebmail-advsearch.pl
--- suidperl/openwebmail-advsearch.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-advsearch.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_advsrch
 #
 # openwebmail-adsearch.pl - advanced search program
 #
diff -ruN suidperl/openwebmail-cal.pl speedycgi/openwebmail-cal.pl
--- suidperl/openwebmail-cal.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-cal.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_cal
 #
 # openwebmail-cal.pl - calendar program
 #
diff -ruN suidperl/openwebmail-folder.pl speedycgi/openwebmail-folder.pl
--- suidperl/openwebmail-folder.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-folder.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_folder
 #
 # openwebmail-folder.pl - mail folder management program
 #
diff -ruN suidperl/openwebmail-main.pl speedycgi/openwebmail-main.pl
--- suidperl/openwebmail-main.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-main.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_main
 #
 # openwebmail-main.pl - message list browsing program
 #
diff -ruN suidperl/openwebmail-prefs.pl speedycgi/openwebmail-prefs.pl
--- suidperl/openwebmail-prefs.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-prefs.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_prefs
 #
 # openwebmail-prefs.pl - preference configuration, book editing program
 #
diff -ruN suidperl/openwebmail-read.pl speedycgi/openwebmail-read.pl
--- suidperl/openwebmail-read.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-read.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_read
 #
 # openwebmail-read.pl - message reading program
 #
diff -ruN suidperl/openwebmail-saprefs.pl speedycgi/openwebmail-saprefs.pl
--- suidperl/openwebmail-saprefs.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-saprefs.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_saprefs
 #
 # openwebmail-saprefs.pl - spamassassin user_prefs file config
 #
diff -ruN suidperl/openwebmail-send.pl speedycgi/openwebmail-send.pl
--- suidperl/openwebmail-send.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-send.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_send
 #
 # openwebmail-send.pl - mail composing and sending program
 #
diff -ruN suidperl/openwebmail-spell.pl speedycgi/openwebmail-spell.pl
--- suidperl/openwebmail-spell.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-spell.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_spell
 #
 # openwebmail-spell.pl - spell check program
 #
diff -ruN suidperl/openwebmail-tool.pl speedycgi/openwebmail-tool.pl
--- suidperl/openwebmail-tool.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-tool.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_tool
 #
 # openwebmail-tool.pl - command tool for mail/event/notify/index...
 #
diff -ruN suidperl/openwebmail-vdomain.pl speedycgi/openwebmail-vdomain.pl
--- suidperl/openwebmail-vdomain.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-vdomain.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_vdomain
 #
 # openwebmail-vdomain.pl - virtual domain user management
 #
diff -ruN suidperl/openwebmail-viewatt.pl speedycgi/openwebmail-viewatt.pl
--- suidperl/openwebmail-viewatt.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-viewatt.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_viewatt
 #
 # openwebmail-viewatt.pl - attachment reading program
 #
diff -ruN suidperl/openwebmail-webdisk.pl speedycgi/openwebmail-webdisk.pl
--- suidperl/openwebmail-webdisk.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail-webdisk.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_webdisk
 #
 # openwebmail-webdisk.pl - web disk program
 #
diff -ruN suidperl/openwebmail.pl speedycgi/openwebmail.pl
--- suidperl/openwebmail.pl	Wed Jan 26 22:29:10 2005
+++ speedycgi/openwebmail.pl	Wed Jan 26 22:42:24 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/suidperl -T
+#!/usr/local/bin/speedy_suidperl -T -- -T/var/run/speedy_login
 #################################################################
 #                                                               #
 # Open WebMail - Provides a web interface to user mailboxes     #
