--- openwebmail-read.pl.orig	Fri Jul 16 19:59:38 2004
+++ openwebmail-read.pl	Fri Jul 16 19:59:23 2004
@@ -816,7 +816,7 @@
 
    my $temphtml .= qq|<table border="0" align="center" cellpadding="2">|.
                    qq|<tr><td valign="middle" bgcolor=$style{"attachment_dark"} align="center">|.
-                   qq|$lang_text{'attachment'} $attnumber: ${$r_attachment}{filename} &nbsp;($attlen)&nbsp;&nbsp;<font color=$style{"attachment_dark"}  size=-2>$nodeid $disposition</font>|.
+                   qq|$lang_text{'attachment'} $attnumber: |.str2html(${$r_attachment}{filename}).qq| &nbsp;($attlen)&nbsp;&nbsp;<font color=$style{"attachment_dark"}  size=-2>$nodeid $disposition</font>|.
                    qq|</td></tr><td valign="middle" bgcolor=$style{"attachment_light"} align="center">|.
                    qq|<a href="$config{'ow_cgiurl'}/openwebmail-viewatt.pl/$escapedfilename?action=viewattachment&amp;sessionid=$thissession&amp;message_id=$escapedmessageid&amp;folder=$escapedfolder&amp;attachment_nodeid=$nodeid$extraparm" title="$lang_text{'download'}">|.
                    qq|<img border="0" |;
@@ -843,13 +843,13 @@
 
    my $temphtml .= qq|<table border="0" width="40%" align="center" cellpadding="2">|.
                    qq|<tr><td nowrap colspan="2" valign="middle" bgcolor=$style{"attachment_dark"} align="center">|.
-                   qq|$lang_text{'attachment'} $attnumber: ${$r_attachment}{filename}&nbsp;($attlen)&nbsp;&nbsp;<font color=$style{"attachment_dark"}  size=-2>$nodeid $disposition|.
+                   qq|$lang_text{'attachment'} $attnumber: |.str2html(${$r_attachment}{filename}).qq|&nbsp;($attlen)&nbsp;&nbsp;<font color=$style{"attachment_dark"}  size=-2>$nodeid $disposition|.
                    qq|</td></tr>|.
                    qq|<tr><td nowrap valign="middle" bgcolor= $style{"attachment_light"} align="center">|.
-                   qq|$lang_text{'type'}: $contenttype<br>|.
-                   qq|$lang_text{'encoding'}: ${$r_attachment}{encoding}|;
+                   qq|$lang_text{'type'}: |.str2html($contenttype).qq|<br>|.
+                   qq|$lang_text{'encoding'}: |.str2html(${$r_attachment}{encoding});
    if (${$r_attachment}{description} ne "") {
-      $temphtml .= qq|<br>$lang_text{'description'}: ${$r_attachment}{description}|;
+      $temphtml .= qq|<br>$lang_text{'description'}: |.str2html(${$r_attachment}{description});
    }
    my $blank="";
    if ($contenttype=~/^text/ ||
