diff -ruN openwebmail-read.pl.orig openwebmail-read.pl
--- openwebmail-read.pl.orig	Wed Jun  2 12:12:30 2004
+++ openwebmail-read.pl	Thu Jun  3 20:27:17 2004
@@ -1117,7 +1117,7 @@
 
    my $temphtml .= qq|<table border="0" align="center" cellpadding="2">|.
                    qq|<tr><td bgcolor=$style{"attachment_dark"} align="center">|.
-                   qq|$lang_text{'attachment'} $attnumber: ${$r_attachment}{filename} &nbsp;($attlen)&nbsp;&nbsp;|;
+                   qq|$lang_text{'attachment'} $attnumber: |.ow::htmltext::str2html(${$r_attachment}{filename}).qq| &nbsp;($attlen)&nbsp;&nbsp;|;
    if ($config{'enable_webdisk'} && !$config{'webdisk_readonly'}) {
       $temphtml .= qq|<a href=#here title="$lang_text{'saveatt_towd'}" onClick="window.open('$config{'ow_cgiurl'}/openwebmail-webdisk.pl?action=sel_saveattachment&amp;sessionid=$thissession&amp;message_id=$escapedmessageid&amp;folder=$escapedfolder&amp;attachment_nodeid=$nodeid$extraparm&amp;attname=|.
                    ow::tool::escapeURL(${$r_attachment}{filename}).qq|', '_blank','width=500,height=330,scrollbars=yes,resizable=yes,location=no');">$lang_text{'webdisk'}</a>|.
@@ -1149,7 +1149,7 @@
 
    my $temphtml .= qq|<table border="0" width="40%" align="center" cellpadding="2">|.
                    qq|<tr><td nowrap colspan="2" bgcolor=$style{"attachment_dark"} align="center">|.
-                   qq|$lang_text{'attachment'} $attnumber: ${$r_attachment}{filename}&nbsp;($attlen)&nbsp;&nbsp;\n|;
+                   qq|$lang_text{'attachment'} $attnumber: |.ow::htmltext::str2html(${$r_attachment}{filename}). qq|&nbsp;($attlen)&nbsp;&nbsp;\n|;
    if (${$r_attachment}{filename}=~/\.(?:doc|dot)$/ ) {
       $temphtml .= qq|<a href="$attlink&amp;wordpreview=1" target="_blank">$lang_wdbutton{'preview'}</a>|.
                    qq|&nbsp;\n|;
@@ -1162,10 +1162,10 @@
    $temphtml .= qq|<font color=$style{"attachment_dark"} class="smalltext">$nodeid $disposition|.
                 qq|</td></tr>|.
                 qq|<tr><td nowrap bgcolor= $style{"attachment_light"} align="center">|.
-                qq|$lang_text{'type'}: $contenttype<br>|.
-                qq|$lang_text{'encoding'}: ${$r_attachment}{'content-transfer-encoding'}|;
+                qq|$lang_text{'type'}: |.ow::htmltext::str2html($contenttype).qq|<br>|.
+                qq|$lang_text{'encoding'}: |.ow::htmltext::str2html(${$r_attachment}{'content-transfer-encoding'});
    if (${$r_attachment}{'content-description'} ne "") {
-      $temphtml .= qq|<br>$lang_text{'description'}: ${$r_attachment}{'content-description'}|;
+      $temphtml .= qq|<br>$lang_text{'description'}: |.ow::htmltext::str2html(${$r_attachment}{'content-description'});
    }
    my $blank="";
    if ($contenttype=~/^text/ ||
