--- image.js	2011-05-25 09:12:48.717321000 -0700
+++ image.js	2011-07-09 01:27:57.000000000 -0700
@@ -492,6 +480,20 @@
 					elements :
 					[
 						{
+							id : 'cmbAttachments',
+							type : 'select',
+							label : 'Attachments',
+                                                       hidden : (editor.config.imageAttachments.unshift([editor.lang.common.notSet,'']) == 1 ? true : false),
+							items : editor.config.imageAttachments,
+							onChange : function()
+							{
+								var dialog = this.getDialog(),
+									newUrl = this.getValue();
+
+								dialog.setValueOf( 'info', 'txtUrl', newUrl );
+							}
+						},
+						{
 							type : 'vbox',
 							padding : 0,
 							children :
