--- openwebmail.conf.orig	Sat Apr 17 03:25:41 2004
+++ openwebmail.conf	Sat Apr 17 03:24:38 2004
@@ -16,11 +16,8 @@
 ow_htmldir		/usr/local/www/data/openwebmail
 ow_htmlurl		/openwebmail
 logfile			/var/log/openwebmail.log
-default_iconset		Default.Chinese.Traditional
 
 <default_signature>
 -- 
-Distributed System Laboratory (http://dslab.ee.ncku.edu.tw)
-Department of Electrical Engineering
-National Cheng Kung University, Tainan, Taiwan, R.O.C.
+Open WebMail Project (http://openwebmail.org)
 </default_signature>
