diff -ruN CGI-SpeedyCGI-2.21.orig/src/SpeedyMake.pl CGI-SpeedyCGI-2.21/src/SpeedyMake.pl
--- CGI-SpeedyCGI-2.21.orig/src/SpeedyMake.pl	Mon Sep 30 13:24:29 2002
+++ CGI-SpeedyCGI-2.21/src/SpeedyMake.pl	Mon May 26 14:52:35 2003
@@ -215,6 +215,7 @@
 	"-DSPEEDY_PROGNAME=\\\"" . $class->my_name_full . "\\\"",
 	"-DSPEEDY_VERSION=\\\"\$(VERSION)\\\"",
 	'-DSPEEDY_' . ($class->am_frontend ? 'FRONTEND' : 'BACKEND'),
+        "-DIAMSUID",
     );
 }
 
diff -ruN CGI-SpeedyCGI-2.21.orig/src/speedy_backend_main.c CGI-SpeedyCGI-2.21/src/speedy_backend_main.c
--- CGI-SpeedyCGI-2.21.orig/src/speedy_backend_main.c	Sat Sep 28 13:16:53 2002
+++ CGI-SpeedyCGI-2.21/src/speedy_backend_main.c	Mon May 26 14:59:27 2003
@@ -176,12 +176,6 @@
     dont_fork = getenv("SPEEDY_NOPARENT") != NULL;
 #endif
 
-    /* Close off all I/O except for stderr (close it later) */
-    for (i = 32; i >= 0; --i) {
-	if (i != 2 && i != PREF_FD_LISTENER)
-	    (void) close(i);
-    }
-
     /*
      * Make sure fd's 0 and 1 are open.
      * Fix for bug where STDOUT couldn't be duped during perl init
@@ -198,6 +192,15 @@
 
     /* Initialize interpreter with this script */
     speedy_perl_init();
+
+    // the following loop is moved from line 178
+    // (the line before  'Make sure fd's 0...') to here 
+    // for suidroot compatibility with Solaris
+    /* Close off all I/O except for stderr (close it later) */
+    for (i = 32; i >= 0; --i) {
+	if (i != 2 && i != PREF_FD_LISTENER)
+	    (void) close(i);
+    }
 
     /* Set up sigs */
     speedy_sig_init(&sl, oursigs, NUM_OURSIGS, SIG_BLOCK);
diff -ruN CGI-SpeedyCGI-2.21.orig/src/speedy_opt.c CGI-SpeedyCGI-2.21/src/speedy_opt.c
--- CGI-SpeedyCGI-2.21.orig/src/speedy_opt.c	Mon Sep 30 13:19:54 2002
+++ CGI-SpeedyCGI-2.21/src/speedy_opt.c	Mon May 26 14:46:00 2003
@@ -165,6 +165,8 @@
 		    ++p;
 	    if (*p)
 		strlist_append(doing_speedy_opts ? speedy_opts : perl_args, *p);
+	    else
+	    	break;
 	}
 
 	if (*p) {
@@ -422,7 +424,10 @@
 }
 
 SPEEDY_INLINE const char *speedy_opt_script_fname(void) {
-    return exec_argv.ptrs[script_argv_loc];
+//    patch for FreeBSD 5.0 or above
+//    return exec_argv.ptrs[script_argv_loc];
+    return exec_argv.len > script_argv_loc ? 
+    		exec_argv.ptrs[script_argv_loc] : NULL;
 }
 
 #ifdef SPEEDY_BACKEND
diff -ruN CGI-SpeedyCGI-2.21.orig/src/speedy_perl.c CGI-SpeedyCGI-2.21/src/speedy_perl.c
--- CGI-SpeedyCGI-2.21.orig/src/speedy_perl.c	Mon Sep 30 13:08:42 2002
+++ CGI-SpeedyCGI-2.21/src/speedy_perl.c	Mon May 26 14:43:47 2003
@@ -31,6 +31,11 @@
 #define pTHXo void
 #endif
 
+/* For perl on RedHat9 */
+#ifndef setdefout 
+#define setdefout(a)     Perl_setdefout(aTHX_ a) 
+#endif
+
 /*
  * Prior to perl 5.6.0 these funcs were prefixed with "perl_"
  */
