--- vacation.pl.orig	Tue Jun 29 08:52:13 2004
+++ vacation.pl	Tue Jun 29 08:56:59 2004
@@ -154,7 +154,7 @@
    }
    my $home = $home_path || $ENV{'HOME'} || (getpwnam($user))[7] or die "No home directory for user $user\n";
    my $editor = $ENV{'VISUAL'} || $ENV{'EDITOR'} || 'vi';
-   my $pager = $ENV{'PAGER'} || 'more';
+   my $pager = 'more'; $pager=$ENV{'PAGER'} if (-f $ENV{'PAGER'});
 
    # guess real homedir under automounter
    $home="/export$home" if ( -d "/export$home" );
@@ -373,6 +373,8 @@
 }
 
 sub read_list_from_file {
+   die ("File $_[0] doesn't exist!\n") if (!-f $_[0]);
+
    my @list=();
    if ( open (FILE, $_[0]) ) {
       while (<FILE>) {
