--- openwebmail.pl.orig	Fri Apr  9 01:49:38 2004
+++ openwebmail.pl	Fri Apr  9 01:51:13 2004
@@ -331,16 +331,6 @@
       }
    }
    upgrade_20030323();
-   # create owuserdir for stuff not put in syshomedir
-   if ( !$config{'use_syshomedir'} || !$config{'use_syshomedir_for_dotdir'} ) {
-      if (!-d $owuserdir) {
-         if (mkdir ("$owuserdir", oct(700)) && chown($uuid, (split(/\s+/,$ugid))[0], $owuserdir)) {
-            writelog("create owuserdir - $owuserdir, uid=$uuid, gid=".(split(/\s+/,$ugid))[0]);
-         } else {
-            openwebmailerror(__FILE__, __LINE__, "$lang_err{'cant_create_dir'} $owuserdir ($!)");
-         }
-      }
-   }
 
    my ($errorcode, $errormsg);
    my $password = param('password') || '';
@@ -365,6 +355,18 @@
 
       # get user release date
       my $user_releasedate=read_releasedatefile();
+
+      # create owuserdir for stuff not put in syshomedir
+      # this must be done before changing to the user's uid.
+      if ( !$config{'use_syshomedir'} || !$config{'use_syshomedir_for_dotdir'} ) {
+         if (!-d $owuserdir) {
+            if (mkdir ($owuserdir, oct(700)) && chown($uuid, (split(/\s+/,$ugid))[0], $owuserdir)) {
+               writelog("create owuserdir - $owuserdir, uid=$uuid, gid=".(split(/\s+/,$ugid))[0]);
+            } else {
+               openwebmailerror(__FILE__, __LINE__, "$lang_err{'cant_create_dir'} $owuserdir ($!)");
+            }
+         }
+      }
 
       # create the user's syshome directory if necessary.
       # this must be done before changing to the user's uid.
